app/guard/guard.go

```diff
@@ -0,0 +1,272 @@
+		},
+		cli.IntFlag{
+			Name:  "debug, d",
+			Value: 0,
+			Usage: "debug-level: `integer`: 1 for terse, 5 for maximal",
+		},
+	}
+	app.Commands = []cli.Command{
+		{
+			Name:    "setpass",
+			Aliases: []string{"s"},
+			Usage:   "Setup the configuration for the server (interactive)",
+			Action:  setpass,
+		},
+		{
+			Name:      "setup",
+			Aliases:   []string{"su"},
+			Usage:     "Saves the cothority group-toml to the configuration",
+			ArgsUsage: "Give group definition",
+			Action:    setup,
+		},
+		{
+			Name:    "recover",
+			Aliases: []string{"r"},
+			Usage:   "Gets the password back from the guards",
+			Action:  get,
+		},
+	}
+	app.Before = func(c *cli.Context) error {
+		b, err := ioutil.ReadFile("config.bin")
+		if os.IsNotExist(err) {
+			return nil
+		}
+		log.ErrFatal(err, "The config.bin file threw an error")
+		_, msg, err := network.UnmarshalRegistered(b)
+		log.ErrFatal(err, "UnmarshalRegistered messeed up")
+		var ok bool
+		db, ok = msg.(*Database)
+		if !ok {
+			log.Fatal("We are not OK")
+		}
+		return nil
+	}
+	app.Run(os.Args)
+}
+
+// readGroup takes a toml file name and reads the file, returning the entities within
+func readGroup(tomlFileName string) (*sda.Roster, error) {
+	log.Print("Reading From File")
+	f, err := os.Open(tomlFileName)
+	if err != nil {
+		return nil, err
+	}
+	el, err := config.ReadGroupToml(f)
+	if err != nil {
+		return nil, err
+	}
+	if len(el.List) <= 0 {
+		return nil, errors.New("Empty or invalid group file:" +
+			tomlFileName)
+	}
+	log.Lvl3(el)
+	return el, err
+}
+
+func set(c *cli.Context, UID []byte, epoch []byte, password string, userdata []byte) {
+	var mastersalt []byte
+	saltsy := make([]byte, 12)
+	rand.Read(saltsy)
+	mastersalt = saltsy
+	k := make([]byte, 32)
+	rand.Read(k)
+	// pwhash is the password hash that will be sent to the guard servers
+	pwhash := abstract.Sum(network.Suite, []byte(password), mastersalt)
+	// secretkeys is the Shamir Secret share of the keys
+	secretkeys := s.Create(2, len(db.Cothority.List), string(k))
+	// salty is a list of salts that will be used to encrypt the pwhash before sending it to the guard, derived from a single
+	// master salt
+	iv := make([]byte, 16)
+	rand.Read(iv)
+	salty := saltgen(mastersalt, len(db.Cothority.List))
+	responses := make([][]byte, len(db.Cothority.List))
+	keys := make([][]byte, len(db.Cothority.List))
+	for i, si := range db.Cothority.List {
+		cl := guard.NewClient()
+		sendy := abstract.Sum(network.Suite, pwhash, salty[i])
+		rep, err := cl.GetGuard(si, UID, epoch, sendy)
+		log.ErrFatal(err)
+		responses[i] = rep.Msg
+		block, err := aes.NewCipher(responses[i])
+		if err != nil {
+			panic(err)
+		}
+		log.Print("Block Size: ", block.BlockSize())
+		// This part is necessary because you need a key to convert the Hash to a stream. The key is not even important because
+		//all that is required is that the request gives the same output for the same input
```

Comment:
Please put a space between the `//` and the text. See http://lkml.iu.edu/hypermail/linux/kernel/1607.1/00627.html